src/NServiceBus.Core/Recoverability/Faults/ErrorQueueSettings.cs

```diff
@@ -23,19 +29,31 @@ public static string ErrorQueueAddress(this ReadOnlySettings settings)
         {
             string errorQueue;
 
-            if (settings.TryGet("errorQueue", out errorQueue))
+            if (TryGetErrorQueueAddress(settings, out errorQueue))
             {
-                Logger.Debug("Error queue retrieved from code configuration via 'EndpointConfiguration.SendFailedMessagesTo().");
                 return errorQueue;
             }
 
+            return DefaultErrorQueueName;
+        }
+
+        internal static bool TryGetErrorQueueAddress(this ReadOnlySettings settings, out string errorQueue)
+        {
+            if (settings.HasExplicitValue(SettingsKey))
+            {
+                Logger.Debug("Error queue retrieved from code configuration via 'EndpointConfiguration.SendFailedMessagesTo().");
+                errorQueue = settings.Get<string>(SettingsKey);
+                return true;
+            }
+
             var section = settings.GetConfigSection<MessageForwardingInCaseOfFaultConfig>();
             if (section != null)
             {
                 if (!string.IsNullOrWhiteSpace(section.ErrorQueue))
                 {
                     Logger.Debug("Error queue retrieved from <MessageForwardingInCaseOfFaultConfig> element in config file.");
-                    return section.ErrorQueue;
+                    errorQueue = section.ErrorQueue;
+                    return true;
```

Comment:
I can't comment directly on the relevant lines, but should we consider revising the two exception messages in this method? One of the possible actions is to just remove the app.config section/registry key, without needing to add the `SendFailedMessagesTo` call (for everything but MSMQ, of course).